### ISSUE TITLE
should I change the value of alpha when using my own data?

### ISSUE THREAD

**Issue by marvin521** [thulas/dac-label-noise#1]
Hi Sunil,  Thank you for contributing to the wonderful code,  I want to apply the method to my own data. Should I change the value of the alpha when performing the method?  It looks like that the method cannot abstain  samples on my own data using default parameter setting (including alpha).

**Comment by thulas** [comment]
Hi,

There are couple of things to try to increase abstention:

1. increase `alpha_init_factor`; try setting it to 128 ( default is 64).
2. Reduce the warm up period (`learn_epochs parameter`). Default is 10 epochs, try 5 epochs.
Hope this helps.

- Sunil.

**Comment by hussam789** [comment]
I'm having the opposite issue, all of the training images are abstained, what `alpha_init_factor` and `alpha_final` you suggest i should use (on larger datasets 1k+ classes)?

**Comment by thulas** [comment]
How many epochs are you training for? Theoretically, if you train for long enough the abstention rate will go to zero.
Anyway, you could  try experimenting with a longer warmup period. Try setting `learn_epochs` to 20, 30 or higher  to see if that helps.

**Comment by pingqingsheng** [comment]
I'm trying to running abstention on cifar10 with noisy level 0.2 using command 

python train_dac.py --dataset cifar10 --datadir ../data --nesterov --net_type resnet2 --depth 34 -use-gpu --epochs 180 -cuda_device 3  --seed 0  --rand_labels 0.2 --loss_fn dac_loss

but I end up with the network abstained all my data points. How may get similar results as the paper suggested ? Are there other parameter to tune ?

Thank you so much !

Songzhu

**Comment by thulas** [comment]
@pingqingsheng: do you mean it abstains on all the training data or validation data, or both? 

In any case, you could try reducing  the `alpha_init_factor` from 64 (default) to 32 or 16. This makes initial abstention less aggressive.

**Comment by pingqingsheng** [comment]
Both.  I reduce it to 2, but I still don't get the ideal number.  It gives 82% on cifar10 with 0.2 noisy rate.  May I know the parameter setting for what you use in the paper?  What is the warm up epochs? Thank you so much!Sent from my Huawei phone-------- Original message --------From: Sunil Thulasidasan <notifications@github.com>Date: Thu, Sep 12, 2019, 4:36 PMTo: thulas/dac-label-noise <dac-label-noise@noreply.github.com>Cc: pingqingsheng <imzszhahahaha@gmail.com>, Mention <mention@noreply.github.com>Subject: Re: [thulas/dac-label-noise] should I change the value of alpha when using my own data? (#1)@pingqingsheng: do you mean it abstains on all the training data or validation data, or both?
In any case, you could try reducing  the alpha_init_factor from 64 (default) to 32 or 16. This makes initial abstention less aggressive.

**Comment by thulas** [comment]
@pingqingsheng 

Try the following:

Step 1:
`python train_dac.py --dataset cifar10 --nesterov --net_type resnet --depth 34 -use-gpu --epochs 165  --loss_fn dac_loss  --learn_epochs 20 --rand_labels 0.2`

Step 2:
Remove the points from the training set that were abstained at the end of training and retrain with regular cross-entropy.

(PS: make sure you save the corrupted indices in step 1 and pass these onto step 2. see --help on how to do this. Otherwise, step 2 will have only clean labels)

**Comment by pingqingsheng** [comment]
Got it. Does the code itself perform the abstained points removing, or we
need to do it ourselves?

**Comment by thulas** [comment]
> Got it. Does the code itself perform the abstained points removing, or we need to do it ourselves?
> […](#)

Currently it does not remove or save a list of  abstained samples; this is a feature I plan to add. For now, you can pass the --save_train_scores option which will save the predictions on the training set at every epoch, using which you can filter out the abstained samples.

**Comment by pingqingsheng** [comment]
I see. Tha't why I don't get the ideal score. So what is the testing score
it gives me finally ?

**Comment by thulas** [comment]
You mean if you retrain on the cleaner (i.e non abstained set)? For CIFAR-10 with ResNet 34 and initially 20% label noise, after retraining you should get about 93.35% on the test set (see Table I from the ICML paper).

**Comment by pingqingsheng** [comment]
I mean in the original code, there is a phase-4 and it will give me a
testing accuracy. What is that accuracy ?

**Comment by thulas** [comment]
Step 1 Phase 4 accuracy will be low (and should be ignored) since many samples are abstained on at the end of training, and abstention label is not in the ground truth. It will  probably be  between 75 and 80%.

 It is Step-2, Phase 4 accuracy -- i.e accuracy after removing noisy points --  that we report in the paper.

**Comment by pingqingsheng** [comment]
Got you.  For step two I don't train with abstained loss right?  Sent from my Huawei phone-------- Original message --------From: Sunil Thulasidasan <notifications@github.com>Date: Fri, Sep 20, 2019, 3:33 PMTo: thulas/dac-label-noise <dac-label-noise@noreply.github.com>Cc: pingqingsheng <imzszhahahaha@gmail.com>, Mention <mention@noreply.github.com>Subject: Re: [thulas/dac-label-noise] should I change the value of alpha when using my own data? (#1)Step 1 Phase 4 accuracy will be low (and should be ignored) since many samples are abstained on at the end of training, and abstention label is not in the ground truth. It will  probably be  between 75 and 80%.
It is Step-2, Phase 4 accuracy -- i.e accuracy after removing noisy points --  that we report in the paper.

**Comment by thulas** [comment]
Yes. Re-train using the regular cross-entropy loss.
